### PR TITLE
tests: block gas limit tests

### PR DESCRIPTION
**Pull request by adisaran64**
Closes: [#1014](https://linear.app/evmos/issue/ENG-1014/blockgaslimit-tests-for-new-antehandler-check) (Linear)

- added block gas limit tests in go tests for ante handler
- added block gas limit tests to integration tests
- added todo for additional functionality (block gas limit checked over sum of tx gas in block)

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: tests/integration_tests/test_gas.py [hunk 1]
```diff
@@ -35,28 +42,68 @@ def test_gas_call(geth, ethermint):
...
+    w3_wait_for_new_blocks(ethermint.w3, 5)
+    block = ethermint.w3.eth.get_block("latest")
+    exceededGasLimit = block.gasLimit + 100
+
+    # send a transaction exceeding the block gas limit
+    ethermint_gas_price = ethermint.w3.eth.gas_price
+    tx = {
+        "to": ADDRS["community"],
+        "value": tx_value,
+        "gas": exceededGasLimit,
```
Comment (fedekunze) [inline tests/integration_tests/test_gas.py:81]:
```suggestion
    exceeded_gas_limit = block.gasLimit + 100

    # send a transaction exceeding the block gas limit
    ethermint_gas_price = ethermint.w3.eth.gas_price
    tx = {
        "to": ADDRS["community"],
        "value": tx_value,
        "gas": exceeded_gas_limit,
```

File: tests/integration_tests/test_gas.py [hunk 2]
```diff
@@ -35,28 +42,68 @@ def test_gas_call(geth, ethermint):
...
+    ethermint_contract, _ = deploy_contract(ethermint.w3, CONTRACTS["BurnGas"])
+
+    # expect an error on contract call due to block gas limit
+    with pytest.raises(Exception):
+        ethermint_txhash = ethermint_contract.functions.burnGas(
+            exceededGasLimit
+        ).transact(
+            {
+                "from": ADDRS["validator"],
+                "gas": exceededGasLimit,
```
Comment (fedekunze) [inline tests/integration_tests/test_gas.py:99]:
```suggestion
            exceeded_gas_limit
        ).transact(
            {
                "from": ADDRS["validator"],
                "gas": exceeded_gas_limit,
```

**Review by GAtom22 — COMMENTED** [review]
Great job! LGTM! For the gas limit tests, should we add a test case when `gas == gasLimit`? To make sure it is working as expected

**Comment by fedekunze** [comment]
@adisaran64 can you reply to @GAtom22's comment?

File: tests/integration_tests/test_gas.py [hunk 3]
```diff
@@ -35,28 +42,68 @@ def test_gas_call(geth, ethermint):
...
+                "gas": exceeded_gas_limit,
+                "gasPrice": ethermint_gas_price,
+            }
+        )
+        (ethermint.w3.eth.wait_for_transaction_receipt(ethermint_txhash))
+
+
+# todo: implement block gas limit being checked for sum of block transactions
+# requires the ability to send batch transactions with web3.py
+def test_many_block_gas_limit(ethermint):
```
Comment (fedekunze) [inline tests/integration_tests/test_gas.py:108]: Can we create an issue for this TODO instead?
```suggestion
```

**Review by fedekunze — APPROVED** [review]
LGTM, pending minor comment and changelog entry

**Comment by facs95** [comment]
https://github.com/evmos/ethermint/issues/1527

**Pull request merged**